src/raqm.c

```diff
@@ -385,27 +399,47 @@ raqm_destroy (raqm_t *rq)
   if (!rq || --rq->ref_count != 0)
     return;
 
-  _raqm_free_intermediate_data (rq);
+  _raqm_release_text_info (rq);
+  _raqm_free_text (rq);
+  _raqm_free_runs (rq);
+  free (rq->glyphs);
+  free (rq->features);
   free (rq);
 }
 
 /**
  * raqm_clear_contents:
  * @rq: a #raqm_t.
+ * @free_memory: whether to free internal buffers or to keep them for reuse.
  *
  * Clears internal state of previously used raqm_t object, making it
  * ready for reuse.
  *
  * Since: 0.9
  */
 void
-raqm_clear_contents (raqm_t *rq)
+raqm_clear_contents (raqm_t *rq,
+                     bool    free_memory)
```

Comment:
I don’t think we need `free_memory`, callers can call `raqm_destroy()` if needed (also since `raqm_t` is reference counted, freeing memory without decreasing the reference count is just wrong.